stytra/experiments/tracking_experiments.py

```diff
@@ -136,30 +163,99 @@ def wrap_up(self, *args, **kwargs):
+    def _setup_frame_dispatcher(self, recording_event: Event = None) -> DispatchProcess:
         """
+        Creates a dispatcher that handles the frames of the camera. It will trigger the recording (i.e. stop it) using
+        the given 'recording_event' event.
 
         Parameters
         ----------
-        exctype :
+        recording_event
+            The event used for recording (if relevant).
+        """
+        return DispatchProcess(
+            self.camera.frame_queue, self.camera.kill_event, recording_event
+        )
 
-        value :
+    def _setup_recording(
+        self, kbit_framerate: int = 1000, extension: str = "mp4"
+    ) -> None:
+        """
+        Does the necessary setup before performing the recording, such as creating events, setting up the dispatcher
+        (via _setup_frame_dispatcher) and initialising the VideoWriter.
 
-        tb :
+        Parameters
+        ----------
+        kbit_framerate
+            the byte rate at which the video is encoded.
+        extension
+            the extension used at the end of the video file.
+        """
+        self.recording_event = Event()
+        self.reset_event = Event()
+        self.finish_event = Event()
 
+        self.frame_dispatcher = self._setup_frame_dispatcher(self.recording_event)
+        self.frame_dispatcher.start()
 
-        Returns
-        -------
+        if extension == "h5":
+            self.frame_recorder = H5VideoWriter(
+                input_queue=self.frame_dispatcher.frame_copy_queue,
+                recording_event=self.recording_event,
+                reset_event=self.reset_event,
+                finish_event=self.finish_event,
+                log_format=self.log_format,
+            )
+        else:
+            self.frame_recorder = StreamingVideoWriter(
+                input_queue=self.frame_dispatcher.frame_copy_queue,
+                recording_event=self.recording_event,
+                reset_event=self.reset_event,
+                finish_event=self.finish_event,
+                kbit_rate=kbit_framerate,
+                log_format=self.log_format,
+            )
 
+        self.frame_recorder.start()
+
+    def _start_recording(self, filename: str) -> None:
         """
+        Pushes the filename to the queue and sets the recording event in order to start the recording.
+
+        Parameters
+        ----------
+        filename
+            a unique identifier that will be added to the video file.
+        """
+        self.frame_recorder.filename_queue.put(filename)
+        self.recording_event.set()
+
+    def _stop_recording(self) -> None:
+        """
+        Stops the recording by clearing the recording event.
+        """
+        self.recording_event.clear()
+
+    def _finish_recording(self) -> None:
+        """
+        Finishes the recording process and joins the frame recorder.
+        """
+        self.frame_recorder.finish_event.set()
+        self.frame_recorder.join()
```

Comment:
I think the recording interface merits its own class, which contains the recording_event and the rest.